GoogleForCreators/web-stories-wp
# Dashboard: Refactor Explore Templates

## Summary
Refactoring explore templates view to prep for real data and functionality 🎉 
Adds unit tests and storybook relevant to explore templates 

## Relevant Technical Choices
- Separates `templates/` into `exploreTemplates` and `templateDetails` 
- Pulls view functionality into a custom hook - mirrors the structure we've got for story side of things 
- Move header and content into separate subdirectories for exploreTemplates
- Adds unit tests and storybook for header and content 

## To-do
- Need to do the same update to the templateDetails view (separate PR) 

## User-facing changes
- New section to storybook Dashboard/Components/exploreTemplates with content and header sections 
- Writing a unit test found one mistake where the search placeholder said 'template stories' 😂  that's fixed now ('search templates') 
- Otherwise, if I did this update right the user facing changes are none. 

## Testing Instructions
- Regression test for the explore templates view - no new functionality
- Ensure new unit tests work
- See that components render accurately in storybook

![explore templates storybook](https://user-images.githubusercontent.com/10720454/83667214-e84e5280-a582-11ea-839c-2d19b18f9328.gif)

---


addresses #2007
